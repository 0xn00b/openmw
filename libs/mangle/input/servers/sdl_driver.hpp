#ifndef MANGLE_INPUT_SDL_DRIVER_H
#define MANGLE_INPUT_SDL_DRIVER_H

#include "../driver.hpp"

namespace Mangle
{
  namespace Input
  {
    /** Input driver for SDL. As the input system of SDL is seldomly
        used alone (most often along with the video system), it is
        assumed that you do your own initialization and cleanup of SDL
        before and after using this driver.

        The Event.event() calls will be given the proper EV_ type, the
        key index (for key up/down events), and a pointer to the full
        SDL_Event structure.
     */
    struct SDLDriver : Driver
    {
      void capture();
      bool isDown(int index);
      void showMouse(bool);
    };
  }
}
#endif
