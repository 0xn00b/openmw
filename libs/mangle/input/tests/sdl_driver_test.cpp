#include "common.cpp"

#include "../servers/sdl_driver.hpp"
#include <SDL.h>

int main(int argc, char** argv)
{
  SDL_Init(SDL_INIT_VIDEO);
  SDL_SetVideoMode(640, 480, 0, SDL_SWSURFACE);
  input = new SDLDriver();

  mainLoop(argc, SDLK_q);

  SDL_Quit();
  return 0;
}
