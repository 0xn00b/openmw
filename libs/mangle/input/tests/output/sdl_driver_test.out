Hold the Q key to quit:
got event: type=1 index=-1
You are running in script mode, aborting. Run this test with a parameter (any at all) to test the input loop properly

Bye bye!
